flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/strategy/AdaptiveBroadcastJoinOptimizationStrategy.java

```diff
@@ -97,92 +102,113 @@ protected void tryOptimizeAdaptiveJoin(
                     adaptiveJoinNode, upstreamEdge.getTypeNumber(), producedBytes);
         }
 
-        // If all upstream nodes have finished, we attempt to optimize the AdaptiveJoin node.
-        if (context.areAllUpstreamNodesFinished(adaptiveJoinNode)) {
-            Long leftInputSize =
-                    aggregatedInputBytesByTypeNumberAndNodeId.get(adaptiveJoinNode.getId()).get(1);
+        FlinkJoinType joinType = adaptiveJoin.getJoinType();
+        Long leftInputSize = null;
+        Long rightInputSize = null;
+
+        // When either input side of the adaptive join meets broadcast requirements, the broadcast
+        // optimization can be immediately applied. This enables maximal utilization of the
+        // {@link ForwardForUnspecifiedPartitioner} for unscheduled nodes on the opposite side,
+        // effectively reducing network overhead.
```

Comment:
// When either input side of the adaptive join meets broadcast requirements, the broadcast 
// optimization can be immediately applied. This allows the opposite side to produce data 
// in a pointwise manner if it has not yet started scheduling, thereby reducing the costs 
// associated with shuffle writes and network overhead. Furthermore, in the future, 
// when multi-input support is available in the runtime, it could enable chaining large input 
// nodes with adaptive join nodes, further minimizing overall overhead.